bin/run-tests.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env sh
+
+# Synopsis:
+# Test the representer by running it against a predefined set of solutions 
+# with an expected output.
+
+# Output:
+# Outputs the diff of the expected representation and mapping against the
+# actual representation and mapping generated by the representer.
+
+# Example:
+# ./bin/run-tests.sh
+
+exit_code=0
+
+# We need to copy the fixtures to a temp directory as the user
+# running within the Docker container does not have permissions 
+# to write to the test_data directory
+test_data_dir="test_data"
+tmp_test_data_dir="/tmp/test_data"
+rm -rf "${tmp_test_data_dir}"
+mkdir -p "${tmp_test_data_dir}"
+cp -R ${test_data_dir}/* "${tmp_test_data_dir}"
+
+# Iterate over all test data directories
+for test_dir in "${tmp_test_data_dir}"/*; do
+    test_dir_name=$(basename "${test_dir}")
+    test_dir_path=$(realpath "${test_dir}")
+
+    bin/run.sh "${test_dir_name}" "${test_dir_path}" "${test_dir_path}"
+
+    file="representation.txt"
+    expected_file="expected_representation.ex"
+    echo "${test_dir_name}: comparing ${file} to ${expected_file}"
+
+    if ! diff "${test_dir_path}/${file}" "${test_dir_path}/${expected_file}"; then
+        exit_code=1
```

Comment:
Can we echo a something about the file being different here? Easier to make sense out of a failure

Reply:
Well, the line above has: 

> echo "${test_dir_name}: comparing ${file} to ${expected_file}"

Wouldn't that be enough?

Reply:
No, because it won't exit here right? It only sets `exit_code=1`, so if something fails we won't know where that happened

Reply:
This is an example of what the output looks like when there is a mismatch:

```
user_defined_structs: comparing representation.txt to expected_representation.ex
2c2
<   defstruct [:wheels, :doors, :color]
---
>   struct [:wheels, :doors, :color]
```

That's not enough you think?

Reply:
Oh, sorry I didn't realize we would get the output from the `diff`. Yes, thank you that's plenty.